dropshot/src/config.rs

```diff
@@ -49,13 +50,35 @@ pub struct ConfigDropshot {
     pub bind_address: SocketAddr,
     /** maximum allowed size of a request body, defaults to 1024 */
     pub request_body_max_bytes: usize,
+
+    // TLS settings
+    /** Whether to use HTTPS. For now, this defaults to false, but we
+     * should think about making it default to true in the future. */
+    pub https: bool,
+    /** Path to a PEM file containing a certificate chain for the
+     *  server to identify itself with. The first certificate is the
+     *  end-entity certificate, and the remaining are intermediate
+     *  certificates on the way to a trusted CA.
+     *
+     *  Only valid if https=true
+     */
+    pub cert_file: PathBuf,
```

Comment:
It seems a little confusing to have values for these properties when TLS is not configured, even if the value is just the empty PathBuf.  Would it make sense for `https` to be an enum with variants like `None` and `Tls(TlsConfig)`, where `TlsConfig` is a struct with `cert_file` and `key_file`?  I wonder if you could do this here with one of the [serde enum tagging modes](https://serde.rs/enum-representations.html).

Reply:
Yeah, agreed. I was having trouble thinking of how to represent this into YAML. I'll dig into the enum tagging stuff

Reply:
Err, TOML, not YAML :) YAML would be rather straightforward

Reply:
I had completely forgotten that TOML has dictionary types, which will likely be the way I'll handle this. I'll probably just do an Option<TlsConfig> for now, since I don't really know of a case for us doing some form of https that isn't TLS

Reply:
Please take a look, I've reworked it